File: hello.js
// REPL for js (on terminal)
// node

// Run js files on terminal
// node <filepath>

// Print
console.log("Hello from Javascript");

// PRIMITIVE TYPES (7)
("Js"); // string
42; // number
3.14; // number
true; // boolean
undefined; // empty
null; // nothing

// Checking types
console.log(typeof 42);
console.log(typeof "Js");

// Casting types
// string => number
// Number.parseInt(<número>, <base decimal>)
console.log(Number.parseInt("42", 10));
// console.log(Number("42"))

// number => string
console.log((42).toString());

// Data structures
js_arr = ["first", 2, true];

js_obj = { name: "alecio", age: 23 };

// VARIABLES and variable naming convention

// var something = "something"

// Variable that won't be re-assigned
// const firstName = "alecio";

// Variable that will be re-assigned
let avatar = "Aang";
avatar += ", the last airbender";

// STRINGS

// length
avatar.length;

// character access
avatar[4];

// substring
console.log(avatar.substring(5, 9));

// Case manipulation
console.log(avatar.toLocaleUpperCase());
avatar.toLocaleLowerCase();

// split
// string => array
teamAvatarString = "Aang, Katara, Sokka";
console.log(teamAvatarString.split(", "));

const firstName = "Toph";
const lastName = "Beifong";

// interpolation
console.log(`${firstName} ${lastName} is the fourth member of team Avatar`);

// ARRAYS
const team_avatar = ["Aang", "Katara", "Sokka"];

// Array CRUD
team_avatar.push("Toph"); // create
team_avatar[3]; // read
team_avatar[2] = "Zuko";
team_avatar.splice(2, 1);

// Loop with foreach
team_avatar.forEach((member) => {
  console.log(member.toUpperCase());
});

// CONTROL FLOW

const bender = "Katara";

if (bender === "Toph") {
  console.log(`${bender} is the best bender!`);
} else if (bender === "Katara") {
  console.log(`${bender} is alsog great, but not the best`);
} else {
  console.log(`Nice try, but no.`);
}

// ternary operator

console.log(bender === "Toph" ? "Best bender!" : "Nice try");

// Falsy values
// null
// undefined
// false
// ""
// 0
// NaN

// OBJECTS
const team_avatar_benders = { aang: "air", katara: "water", toph: "earth" };

// reading a property
console.log(team_avatar_benders["aang"]);
console.log(team_avatar_benders.katara);

// setting a property
team_avatar_benders["zuko"] = "fire";
// team_avatar_benders.zuko = 'fire';
// console.log(team_avatar_benders);

// iterating through an object
console.log(Object.values(team_avatar_benders));
Object.entries(team_avatar_benders).forEach(([member, element]) => {
  console.log(`${member} bends ${element}`);
});

// FUNCTIONS

// function statement

// function introducePet(name) {
//   console.log(`${name} is an awesome pet`);
// }

// function expression
const introducePet = function (name) {
  console.log(`${name} is an awesome pet`);
};
introducePet("Appa");

// arrow function
const introducePetArrow = (name) => {
  console.log(`${name} is an awesome pet`);
};
introducePetArrow("Momo");

// Capitalize example

const capitalize = (word) => {
  const firstLetter = word[0].toUpperCase();
  const restOfTheWord = word.slice(1).toLowerCase();
  return `${firstLetter}${restOfTheWord}`;
};

console.log(capitalize("MOMO"));

File: hello.rb
# REPL for ruby (on terminal)
# irb





# Run ruby files on terminal
# ruby <filepath>





# Print
puts "Hello from Ruby"






# RUBY TYPES
"Ruby" # String
42     # Integer
3.14   # Float
true   # TrueClass
nil    # NilClass





# Checking types
puts 42.class
puts true.class





# Casting types
# string => number
"42".to_i





# number => string
42.to_s





# Data structures
ruby_arr = ["first", 2, true]





ruby_obj = { name: 'alecio', age: 23 }
ruby_obj2 = { 'name' => 'alecio', 'age' => 23 }





# VARIABLES

avatar = "Aang"
avatar += ", the last airbender"




# STRINGS

# length
avatar.size





# character access
avatar[6]





# substring
avatar.slice(6..9)





# Case manipulation
avatar.upcase
avatar.downcase





# split
# string => array
team_avatar_string = "Aang, Katara, Sokka"
avatar.split(", ")

first_name = "Toph"
last_name = "Beifong"
# interpolation
puts "#{first_name} #{last_name} is the fourth member of team avatar"


# ARRAYS
team_avatar = ['Aang', 'Katara', 'Sokka']

# Array CRUD
team_avatar << 'Toph'
team_avatar[1]
team_avatar[2] = 'Zuko'
team_avatar.delete_at(2)

# Loop with each
team_avatar.each do |member|
  puts member.upcase
end 


# CONTROL FLOW

bender = 'Toph'

if bender == 'Toph'
  puts "#{bender} is the best bender!"
elsif
  puts "#{bender} is also great, but not the best."
else
  puts "Nice try, but no"
end

# ternary operator
puts bender == 'Toph' ? "Best bender!" : "Nice try"

# Falsy values
# nil
# false


# OBJECTS
team_avatar_benders = { aang: 'air', katara: 'water', toph: 'earth' }

# reading a property
team_avatar_benders[:aang]

# setting a property
team_avatar_benders[:zuko] = 'fire'

# iterating through an object
team_avatar_benders.each do |member, element|
  puts "#{member} bends #{element}"
end 


# FUNCTIONS
def introduce_pet(name)
  puts "#{name} is an awesome pet"
end

introduce_pet('Appa')

